File: frontend/src/dto/branch.dto.ts
import { UUID } from "crypto";

export interface BranchDTO {
    branchId: UUID;
    restaurantId: UUID;
    address: string;
    branchPhone?: string;
    openingTime?: string;
    closingTime?: string;
    isActive: boolean;
    mail?: string;
}

export interface CreateBranchDTO {
    restaurantId: UUID;
    address: string;
    branchPhone?: string;
    openingTime?: string;
    closingTime?: string;
    isActive?: boolean;
    mail?: string;
}

export interface UpdateBranchDTO {
    address?: string;
    branchPhone?: string;
    openingTime?: string;
    closingTime?: string;
    isActive?: boolean;
    mail?: string;
}

File: backend/src/main/java/com/example/backend/dto/BranchDTO.java
package com.example.backend.dto;

import java.time.LocalTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

public class BranchDTO {
    private UUID branchId;
    private UUID restaurantId;
    private String address;
    private String branchPhone;
    private LocalTime openingTime;
    private LocalTime closingTime;
    @JsonProperty("isActive")
    private boolean active = true;
    private String mail;

    public UUID getBranchId() {
        return branchId;
    }

    public void setBranchId(UUID branchId) {
        this.branchId = branchId;
    }

    public UUID getRestaurantId() {
        return restaurantId;
    }

    public void setRestaurantId(UUID restaurantId) {
        this.restaurantId = restaurantId;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getBranchPhone() {
        return branchPhone;
    }

    public void setBranchPhone(String branchPhone) {
        this.branchPhone = branchPhone;
    }

    public LocalTime getOpeningTime() {
        return openingTime;
    }

    public void setOpeningTime(LocalTime openingTime) {
        this.openingTime = openingTime;
    }

    public LocalTime getClosingTime() {
        return closingTime;
    }

    public void setClosingTime(LocalTime closingTime) {
        this.closingTime = closingTime;
    }

    public boolean isActive() {
        return active;
    }

    @JsonProperty("isActive")
    public void setActive(boolean active) {
        this.active = active;
    }

    public String getMail() {
        return mail;
    }

    public void setMail(String mail) {
        this.mail = mail;
    }
}
